Rocket.Chat/Controllers/Chat/ChatViewController.swift

```diff
@@ -116,6 +120,21 @@ final class ChatViewController: SLKTextViewController {
         }
     }
 
+    override func viewWillAppear(_ animated: Bool) {
+        super.viewWillAppear(animated)
+
+        let menuItemFilter: (UIBarButtonItem) -> Bool = { barButtonItem in
+            if let button = barButtonItem.customView as? UIButton,
+                let sideMenuController = self.sideMenuController {
+                return button.allTargets.contains(sideMenuController)
+            }
+            return false
+        }
+        guard let leftMenuButtonItem = self.navigationItem.leftBarButtonItems?.filter(menuItemFilter).first else { return }
+
+        leftMenuButtonItem.addBadge(self.menuBadge)
```

Comment:
Well, that's too messy, in my opinion. I think the view controller should know much fewer the better. And we should try not to call a library's private methods. A simple patch is good enough.